### ISSUE TITLE
How to load older messages

### ISSUE THREAD

**Issue by divyabk1** [pazznetwork/ngx-chat#92]
Hello..Can you pls let me know how to load archive messages in ngx-chat?

**Comment by juergen83** [comment]
Use the Pugin designed for that:

`this.chatService.getPlugin(MessageArchivePlugin).loadAllMessages()`

**Comment by divyabk1** [comment]
On what event can I invoke this method?

**Comment by juergen83** [comment]
Inside the component 'ChatMessageListComponent' you find this plugin triggered by Lifecycle Hook 'ngAfterViewInit':
`async ngAfterViewInit() {
...
if (this.recipient.messages.length < 10) {
      await this.loadMessages(); // in case insufficient old messages are displayed
}
...
}

private async loadMessages() {
        try {
            // improve performance when loading lots of old messages
            this.changeDetectorRef.detach();
            await this.chatService.getPlugin(MessageArchivePlugin).loadMostRecentUnloadedMessages(this.recipient);
        } finally {
            this.changeDetectorRef.reattach();
        }
    }
`

**Comment by divyabk1** [comment]
In my component, should I invoke this.chatMessageListComponent.loadOlderMessagesBeforeViewport(); ?

**Comment by juergen83** [comment]
If your component extends the ChatMessageListComponent then you can use this method.
But if you extends this Component you have to overwrite all (used) private variables in it.

**Comment by divyabk1** [comment]
What is the other way to do it?
